qemu/tests/bridge_vlan.py

```diff
@@ -0,0 +1,239 @@
+        :params: vlan_if: Vlan interface.
+        :params: mac_str: New mac address for vlan.
+        :params: session: VM session.
+        """
+        mac_cmd = "ip link set %s add %s up" % (vlan_if, mac_str)
+        error_context.context("Give a new mac address '%s' for vlan interface '%s'" %
+                              (mac_str, vlan_if), logging.info)
+        session.cmd(mac_cmd)
+
+    def set_arp_ignore(session):
+        """
+        Enable arp_ignore for all ipv4 device in guest
+        """
+        error_context.context("Enable arp_ignore for all ipv4 device in guest",
+                              logging.info)
+        ignore_cmd = "echo 1 > /proc/sys/net/ipv4/conf/all/arp_ignore"
+        session.cmd(ignore_cmd)
+
+    def ping_vlan(vm, dest, vlan_if, session):
+        """
+        Test ping between vlans, from guest to host/guest.
+        :params vm: VM object
+        :params dest: Dest ip to ping.
+        :params vlan_if: Vlan interface.
+        :params session: VM session.
+        """
+        error_context.context("Test ping from '%s' to '%s' on guest '%s'" %
+                              (vlan_if, dest, vm.name))
+        status, output = utils_test.ping(dest=dest, count=10,
+                                         interface=vlan_if,
+                                         session=session,
+                                         timeout=30)
+        if status:
+            err_msg = "%s ping %s unexpected, " % (vm.name, dest)
+            err_msg += "error info: %s" % output
+            test.fail(err_msg)
+
+    def netperf_vlan(client='main_vm', server='localhost', sub_type='netperf_stress'):
+        """
+        Test netperf stress among guests and host.
+        :params client: Netperf client.
+        :params server: Netperf server.
+        :params sub_type: Sub_type to run.
+        """
+        params["netperf_client"] = client
+        params["netperf_server"] = server
+        error_context.context("Run netperf stress test among guests and host, "
+                              "server: %s, client: %s" % (server, client),
+                              logging.info)
+        utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+    vms = []
+    sessions = []
+    ifname = []
+    vm_ip = []
+    vm_vlan_ip = []
+    vm_vlan_if = []
+    sub_type = params["sub_type"]
+    host_vlan_id = params.get("host_vlan_id", "10")
+    host_vlan_ip = params.get("host_vlan_ip", "192.168.10.10")
+    subnet = params.get("subnet", "192.168")
+    mac_str = params.get("mac_str").split(',')
+    os_type = params.get("os_type", "linux")
+    login_timeout = int(params.get("login_timeout", 360))
+
+    linux_modules.load_module("8021q")
+
+    host_vlan_if = "%s.%s" % ("switch", host_vlan_id)
+    if host_vlan_if not in utils_net.get_net_if():
+        host_vlan_if = add_vlan(interface="switch", v_id=host_vlan_id)
+        if host_vlan_if and host_vlan_if in utils_net.get_net_if():
```

Comment:
@sitoliu  "if host_vlan_if" seems doesn't work here, as it's can not be empty or none.